pkg/minikube/cluster/cluster.go

```diff
@@ -199,6 +200,10 @@ func engineOptions(config MachineConfig) *engine.Options {
 
 func createVirtualboxHost(config MachineConfig) drivers.Driver {
 	d := virtualbox.NewDriver(constants.MachineName, constants.Minipath)
+	if err := minishiftUtil.SetEnvDriverOptions(d); err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error in setting config from flags: %s", err.Error()))
```

Comment:
Not sure whether it is a good idea to use atexit on this level. I would think atexit is something used from the cmd part of Minishift. Here on the pkg level we should stick to the Go way of doing things and return an error. By calling Exit you terminate the program here and now. What if some cleanup is needed higher up. So we need to change the signature here and return an error.